store/tikv/oracle/oracles/pd.go

```diff
@@ -231,3 +238,47 @@ func (o *pdOracle) GetLowResolutionTimestampAsync(ctx context.Context, opt *orac
 		err: nil,
 	}
 }
+
+func (o *pdOracle) getStaleTimestamp(ctx context.Context, prevSecond int64) (uint64, error) {
+	ts, ok := o.getLastTS(oracle.GlobalTxnScope)
+	if !ok {
+		return 0, errors.Errorf("get stale timestamp, invalid txnScope = %s", oracle.GlobalTxnScope)
+	}
+	tsArrival, ok := o.getLastArrivalTS()
+	if !ok {
+		return 0, errors.Errorf("get last arrival timestamp, invalid txnScope = %s", oracle.GlobalTxnScope)
+	}
+	arrivalTime := oracle.GetTimeFromTS(tsArrival)
+	physicalTime := oracle.GetTimeFromTS(ts)
+	if physicalTime.Unix() <= prevSecond {
+		return 0, errors.Errorf("get invalid prevSecond, prevSecond must less than physicalTime, "+
+			"get prevSecond: %d, physicalTime: %d", prevSecond, physicalTime.Unix())
+	}
+
+	staleTime := physicalTime.Add(-arrivalTime.Sub(time.Now().Add(-time.Duration(prevSecond) * time.Second)))
+
+	return oracle.ComposeTS(oracle.GetPhysical(staleTime), 0), nil
+}
+
+// GetStaleTimestamp generate a TSO which represents for the TSO prevSecond secs ago.
+func (o *pdOracle) GetStaleTimestamp(ctx context.Context, prevSecond int64) (ts uint64, err error) {
+	ts, err = o.getStaleTimestamp(ctx, prevSecond)
+	if err != nil {
+		return 0, errors.Wrap(err, "")
```

Comment:
```suggestion
		return 0, errors.Trace(err)
```

Reply:
Thanks for your review, I agree with you.